### ISSUE TITLE
Managing Dependencies is a Big question

### ISSUE THREAD

**Issue by ivansky** [teambit/bit#6127]
### Description
There are some questions I'm asking myself for more than a year working with bit, it would be awesome to have some answers or to have a strong opinion on that matter.

1. How to make sure that all projects with different `workspace.jsonc` will not override dependencies and it will not make tag juggling (Project a -> b -> a -> b -> ....). Any difference in a version leads to having new change set.
1. `workspace.jsonc` has `dependencies`, `peerDependencies` and `overrides` (is it `resolutions` thing?). Why it doesn't use `package.json` to store dependencies, so we could keep any IDE default support for dependencies (auto-completion, ...)
1. Bit tries to be a package manager, but it does not provide the full power of a PM, e.g. npm/yarn/etc..., so the things like running scripts are missing, so we need to have any PM attached to a project. What it should be? Should it be the same as mentioned in `workspace.jsonc`? Should it be the same version? Could it share cache? 
1. Why after `bit install` that includes yarn berry inside, the yarn berry in the project when it's called to run a script, it might fail with a message that packages are corrupted?
1. Why bit takes into account packages mentioned in `package.json`, but when it's `bit update` it fully ignores it?

### Describe the solution you'd like
I would like to:

1. have dependencies in one place that is a `package.json`
1. be able to `bit update` dependencies stored in `package.json` not only in `workspace.jsonc`
1. be sure that the `bit update` always affects installed packages (sometimes it fails because the version in lock file or package.json suits the installed one and `bit update` only changes number in `workspace.jsonc`)

**Comment by hyoretsu** [comment]
He raises some huge points. I was looking to fully switch to Bit, but I'll need to have a package.json because I need custom npm scripts.

**Comment by sbland** [comment]
Should this be moved to discussions?

After also using Bit for over a year now I've found it is trying to be to replace too many systems at once without resolving some of the root problems first or providing easy integration with established systems. 

From my understanding Bit is...
 - A version control system
 - A Package Manager
 - A UI "workshop" I.e. like storybook
 - A deployment framework

Another Dependencies related issue I have came across is the following situation...

Component A(V1.0.0) depends on Component B(V1.0.0) which depends on component C(V1.0.0).

If component C has a braking change its version changes to `V1.1.0`. Bit would then automatically update Component A and B to `V1.1.0`. Any dependencies on Component A would then assume there is a braking change. But if the braking change is handled in Component B without any change to Component B's API then there is no braking change to Component A.
